client/extensions/woocommerce/app/products/products-list-search-results.js

```diff
@@ -87,9 +87,13 @@ ProductsListSearchResults.propTypes = {
 function mapStateToProps( state ) {
 	const site = getSelectedSiteWithFallback( state );
 	const currentPage = site && getProductSearchCurrentPage( state, site.ID );
-	const currentPageLoaded = site && currentPage && areProductSearchResultsLoaded( state, currentPage, site.ID );
+	const currentPageLoaded = site && currentPage && areProductSearchResultsLoaded(
+		state, { page: currentPage, per_page: 10 }, site.ID
```

Comment:
I think this, and the `requestedPageLoaded` change below both need to also pass the current search query string. 

Searching works in my testing, but I'm not getting the correct `No products match your search` message when there are no results. I think that is because of [this check](https://github.com/Automattic/wp-calypso/blob/ea5a409b31b49bb241a633e175d77b4c285ba9e0/client/extensions/woocommerce/app/products/products-list-search-results.js#L38) which would use the value here. But the actual string stored in state is:

`{"page":1,"per_page":10,"search":"search term"}: false`

Reply:
Fixed in cddd8c6